### PR TITLE
LPD-43990 GET operation for a Content Page

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPD-43990

This PR specifically targets LPD-43990, but it also incorporates commits from [LPD-44470 and LPD-44481 Implement Basic POST operations for Page Specification Part 2](https://github.com/liferay-page-management/liferay-portal/pull/16759) to prevent merge conflicts.

### CODE REVIEW

**Comment by liferay-page-management** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/dto/v1_0/converter/SitePageDTOConverter.java [hunk 1]
```diff
@@ -60,18 +76,106 @@ public SitePage toDTO(
...
+		}
+
+		if (Objects.equals(
+				collectionType,
+				InfoListProviderItemSelectorReturnType.class.getName())) {
+
+			return new ClassNameReference() {
+				{
+					setClassName(
+						() -> layout.getTypeSettingsProperty("collectionPK"));
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/dto/v1_0/converter/SitePageDTOConverter.java:152]: @lfbesada Is collectionPK the right property to read here?

Comment (lfbesada) [reply modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/dto/v1_0/converter/SitePageDTOConverter.java:152]: I guess so as the ClassNameReference object only has the className field and the collectionPK should be exported somehow.

Here are examples of collection layout's type settings
- with a collection provider
```
collectionPK=com.liferay.asset.internal.info.collection.provider.RecentContentInfoCollectionProvider
collectionType=com.liferay.info.list.provider.item.selector.criterion.InfoListProviderItemSelectorReturnType

```
- with an asssetListEntry
```
collectionPK=128715
collectionType=com.liferay.item.selector.criteria.InfoListItemSelectorReturnType
```

**Comment by lfbesada** [comment]
@ruben-pulido I only rebased with the current master, with no changes. Thanks for your review :heart:

**Comment by lfbesada** [comment]
I only rebased with the current master to solve conflicts on imports, with no changes.

**Pull request closed without merging**